tools/package-macos-arm64.sh

```diff
@@ -21,7 +21,13 @@ else
   DART=$(which dart)
 fi
 
-eval "$DART pub get -C $SRC_DIR"
+restore_backup_and_exit() {
+  mv "$SRC_DIR/pubspec_overrides.back.yaml" "$SRC_DIR/pubspec_overrides.yaml"
+  exit "$1"
+}
+
+mv "$SRC_DIR/pubspec_overrides.yaml" "$SRC_DIR/pubspec_overrides.back.yaml"
+eval "$DART pub get -C $SRC_DIR" || restore_backup_and_exit 1
```

Comment:
This prevents accidentally packaging for apple-silicon locally if you've run melos bootstrap on your system.